### DISCUSSION TITLE
WDT help for Raspberry Pico

### DISCUSSION THREAD

**Discussion by bzisl** [earlephilhower/arduino-pico#997] [Q&A] [answered]
Hi! I'm trying to use the WDT to allow a reboot in case of firmware hang, but there is no examples, or the doc is not available.
Which header we myst include?
Can someone help?

Thanks in advance.

**Comment by Andy2No** [comment, marked as answer]
There's a 50 minute lecture on the subject, by Van Hunter Adams.  I haven't watched it yet, but they're always worth watching.  He's made a whole series on the Pico, in recent times:

https://www.youtube.com/watch?v=dTfkopqqFH8

**Reply by bzisl** [reply]
Thanks! I will see now!

**Comment by earlephilhower** [comment]
https://arduino-pico.readthedocs.io/en/latest/rp2040.html#hardware-watchdog has the built in calls.  Note that the doc needs a minor ediot, the parameter really is **milli**seconds, not microseconds.  You can also use the SDK functions bare with `#include "hardware/watchdog.h"`

**Reply by bzisl** [reply]
Thanks!!!
